client/my-sites/themes/eligibility-warning-modal.tsx

```diff
@@ -0,0 +1,95 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { Dialog } from '@automattic/components';
+import { localize } from 'i18n-calypso';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import EligibilityWarnings from 'calypso/blocks/eligibility-warnings';
+import {
+	acceptEligibilityWarningDialog,
+	dismissEligibilityWarningDialog,
+	activate as activateTheme,
+} from 'calypso/state/themes/actions';
+import {
+	getCanonicalTheme,
+	getThemeForEligibilityWarning,
+	shouldShowEligibilityWarning,
+} from 'calypso/state/themes/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+import './auto-loading-homepage-modal.scss';
+
+interface Props {
+	siteId?: number;
+	showEligibility: boolean;
+	theme: {
+		author: string;
+		author_uri: string;
+		id: string;
+		name: string;
+	};
+	dispatchAcceptEligibilityWarningDialog: typeof acceptEligibilityWarningDialog;
+	dispatchDismissEligibilityWarningDialog: typeof dismissEligibilityWarningDialog;
+	dispatchActivateTheme: typeof activateTheme;
+	dispatchRecordTracksEvent: typeof recordTracksEvent;
+}
+class EligibilityWarningModal extends Component< Props > {
```

Comment:
Thought: it's not obvious to me what `Eligibility` this is about. Might it be clearer to call this something like `AtomicTransferDialog`? It might be more flexible if we also pass in a callback like `onAcceptAtomicTransfer` and any descriptive text as props. (We might also have such a component already.)